fiftyone/utils/utils3d.py

```diff
@@ -0,0 +1,204 @@
+"""
+3D utilities.
+
+| Copyright 2017-2022, Voxel51, Inc.
+| `voxel51.com <https://voxel51.com/>`_
+|
+"""
+import numpy as np
+import scipy.spatial as sps
+
+import eta.core.numutils as etan
+
+
+# Reference: https://github.com/AlienCat-K/3D-IoU-Python/blob/master/3D-IoU-Python.py
+def compute_cuboid_iou(gt, pred, gt_crowd=False):
+    """Computes the IoU between the given ground truth and predicted cuboids.
+
+    Args:
+        gt: a :class:`fiftyone.core.labels.Detection`
+        pred: a :class:`fiftyone.core.labels.Detection`
+        gt_crowd (False): whether the ground truth cuboid is a crowd
+
+    Returns:
+        the IoU, in ``[0, 1]``
+    """
+    corners1 = _3d_corners(gt)
+    corners2 = _3d_corners(pred)
+
+    # points are in counter-clockwise order
+    rect1 = [(corners1[i, 0], corners1[i, 2]) for i in range(3, -1, -1)]
+    rect2 = [(corners2[i, 0], corners2[i, 2]) for i in range(3, -1, -1)]
+
+    _, inter_area = _convex_hull_intersection(rect1, rect2)
+    ymax = min(corners1[0, 1], corners2[0, 1])
+    ymin = max(corners1[4, 1], corners2[4, 1])
+    inter_vol = inter_area * max(0.0, ymax - ymin)
+
+    # area1 = _polygon_area(np.array(rect1)[:, 0], np.array(rect1)[:, 1])
+    # area2 = _polygon_area(np.array(rect2)[:, 0], np.array(rect2)[:, 1])
+    # iou_2d = inter_area / (area1 + area2 - inter_area)
+
+    gt_vol = _cuboid_volume(corners1)
+    pred_vol = _cuboid_volume(corners2)
+
+    if gt_crowd:
+        union = pred_vol
+    else:
+        union = gt_vol + pred_vol - inter_vol
+
+    return min(etan.safe_divide(inter_vol, union), 1)
+
+
+def _3d_corners(detection):
+    rotation = np.array(detection.rotation)
+    location = np.array(detection.location)
+    x, y, z = np.array(detection.dimensions)
+    R = _rotation_matrix(rotation)
+
+    corners = np.vstack(
+        [
+            [x / 2, x / 2, -x / 2, -x / 2, x / 2, x / 2, -x / 2, -x / 2],
+            [z / 2, z / 2, z / 2, z / 2, -z / 2, -z / 2, -z / 2, -z / 2],
+            [y / 2, -y / 2, -y / 2, y / 2, y / 2, -y / 2, -y / 2, y / 2],
+        ]
+    )
+
+    return (R @ corners + location[:, np.newaxis]).T
+
+
+def _rotation_matrix(rotation):
```

Comment:
This implementation matches what `o3d.geometry.get_rotation_matrix_from_xyz(rotation)` returns, which is what I used when generating the toy dataset in https://raw.githubusercontent.com/voxel51/fiftyone/develop/docs/source/user_guide/groups.rst.

Saying this because the toy dataset proves that this order of rotation matches how Looker3D renders 3D detections in the App.